forms/Form.php

```diff
@@ -1416,10 +1416,13 @@ public function loadDataFrom($data, $mergeStrategy = 0, $fieldList = null) {
 					// First encode data using PHP's method of converting nested arrays to form data
 					$flatData = urldecode(http_build_query($data));
 					// Then pull the value out from that flattened string
-					preg_match('/' . addcslashes($name,'[]') . '=([^&]*)/', $flatData, $matches);
+					preg_match_all('/' . addcslashes($name,'[]') . '[^=]*=([^&]*)/', $flatData, $matches);
 
-					if (isset($matches[1])) {
+					if (isset($matches[1]) && sizeof($matches[1]) == 1) {
 						$exists = true;
+						$val = $matches[1][0];
```

Comment:
This isn't strictly correct is it? What if it's an associative array, and the key of the first element isn't a strict '0'?